src/bot/snoopiBot.jl

```diff
@@ -0,0 +1,109 @@
+
+################################################################
+"""
+    @snoopiBot config::BotConfig snoopScript
+
+macro that generates precompile files and includes them in the package. Calls other bot functions.
+
+# Examples
+
+`@snoopiBot` the examples that call the package functions.
+
+```julia
+using SnoopCompile
+
+@snoopiBot "MatLang" begin
+  using MatLang
+  examplePath = joinpath(dirname(dirname(pathof(MatLang))), "examples")
+  include(joinpath(examplePath,"Language_Fundamentals", "usage_Entering_Commands.jl"))
+  include(joinpath(examplePath,"Language_Fundamentals", "usage_Matrices_and_Arrays.jl"))
+  include(joinpath(examplePath,"Language_Fundamentals", "Data_Types", "usage_Numeric_Types.jl"))
+end
+```
+"""
+macro snoopiBot(config::BotConfig, snoopScript)
+
+    packageName = config.packageName
+    blacklist = config.blacklist
+    subst = config.subst
+    ################################################################
+    packagePath = joinpath(pwd(),"src","$packageName.jl")
+    precompilePath, precompileFolder = precompilePather(packageName)
+
+    quote
+        packageSym = Symbol($packageName)
+        ################################################################
+        using SnoopCompile
+        ################################################################
+        precompileDeactivator($packagePath, $precompilePath);
+        ################################################################
+
+        ### Log the compiles
+        data = @snoopi begin
+            $(esc(snoopScript))
+        end
+
+        ################################################################
+        ### Parse the compiles and generate precompilation scripts
+        pc = SnoopCompile.parcel(data, subst = $subst, blacklist = $blacklist)
+        onlypackage = Dict( packageSym => sort(pc[packageSym]) )
```

Comment:
This `sort` isn't necessary anymore I think

Reply:
So does parcel sort the signatures already? 

The reason I have this is to avoid creating pull requests that just change the order of lines.

Reply:
Actually, only `write` does this: https://github.com/timholy/SnoopCompile.jl/blob/95103414dc870dfab0d46118dcb9c050b038c5fe/src/write.jl#L38

So keep it if you need it.